### ISSUE TITLE
Fix DMs.

### ISSUE THREAD

**Issue by ara4n** [vector-im/riot-web#7315]
We have hopes and dreams of simplifying DMs:
 * Making them symmetrical for both users (i.e. decorating the room, not the user's view of the room)
 * Making them immutable
and stuff.

Placeholder to write that up.

**Comment by ara4n** [comment]
fix in the redesign

**Comment by ara4n** [comment]
i've lost my notes for this >:(

**Comment by ara4n** [comment]
I found my notes, which i'm going to c+p verbatim here before i lose them again. (these are not meant to make sense to anyone but me and possibly @AmandineLP; they're notes taken from arguing this through as we flew to DWS 2018 iirc):

Things to fix:
 * Everyone should have the same view of the room.
 * Stop confusing users by letting them feel they can invite users into a DM (but we support it for bots and super-powerusers anyway).  The proto should support it, but we can hide it from the UX by default.
 * Fractal should be able to ditch public rooms in the BBQ view, if it wants to.

Options:
 1. Private v. Public  (i.e. 1:1s and Priv Group chat v. Public) where public is join_rule based
 2. DMs v. Private Rooms v. Public Rooms  (where DMs can include bots)
 3. DMs v Rooms
 4. Define groupness based on notif settings?  (doesn't make sense; everyone should have the same view of the room).

M&A agree option 2 is simplest.
DMs are immutable.

What about private groups?
  * Obviously public rooms should be able to be join_ruled invite only when needed; but do they move to private rooms or not?
  * Either: no: because the intent was clearly that they were public and they shouldn't jump around the menu on join-rule changes
  * Or... yes: because we should put the room in the right category.
    * => we vote yes.

---------------

After discussing with Tobias on private group immutability:
 * This would be weird for the Slack folks, as company group chats would get entangled with private group chats.
 * In other words, publicness is relative.

What makes a channel a channel and not a group?
 * Public join rules?
 * Join rules available to a group?
 * An alias?  <-- this could be the best bet, but given anyone can add an alias... >:(
 * A name?
 * The intent when it is created?
 * The intent at any point?

**Comment by ara4n** [comment]
TL;DR: current conclusion is:
 * make DMs immutable
 * ensure there's only one DM room for a given pair of users, tracking on room state (somehow)
 * additional users (e.g. bots) can still be invited into DMs though.

In future:
 * split rooms based on join_rules into "private groups" and "public rooms" or something.
   * perhaps call DMs "private chats"?
 * as join_rules are toggled on and off for a room, they can move between these two sections. i.e. these are *not* immutable.
 * if you want to separate your workplace private groups from your social private groups, use communities.
 * remove "low priority" from the LeftPanel, hiding it as an option in the TopLeftMenu, to make room for "private groups".
